aspnetcore/performance/performance-best-practices.md

```diff
@@ -0,0 +1,136 @@
+
+* Make [hot code paths](#hot) asynchronous.
+* Call asynchronous APIs for any long-running operations.  
+* Call  data access APIs asynchronously.
+* Make controller/Razor Page actions asynchronous. The entire call stack needs to be asynchronous in order to benefit from [async/await](/dotnet/csharp/programming-guide/concepts/async/) patterns.
+
+A profiler like [PerfView](https://github.com/Microsoft/perfview) can be used to look for threads frequently being added to the [Thread Pool](/windows/desktop/procthread/thread-pool). The `Microsoft-Windows-DotNETRuntime/ThreadPoolWorkerThread/Start` event indicates a thread being added to the thread pool. <!--  For more information, see [async guidance docs](TBD-Link_To_Davifowl_Doc  -->
+
+## Minimize large object allocations
+
+<!-- TODO review Bill - replaced original .NET language below with .NET Core since this targets .NET Core -->
+The [.NET Core garbage collector](/dotnet/standard/garbage-collection/) manages allocation and release of memory automatically in ASP.NET Core apps. Automatic garbage collection generally means that developers don't need to worry about how or when memory is freed. However, cleaning up unreferenced objects takes CPU time, so developers should minimize allocating objects in [hot code paths](#hot). Garbage collection is especially expensive on large objects (> 85K bytes). Large objects are stored on the [large object heap](/dotnet/standard/garbage-collection/large-object-heap) and require a full (generation 2) garbage collection to clean up. Unlike generation 0 and generation 1 collections, a generation 2 collection requires app execution to be temporarily suspended. Frequent allocation and de-allocation of large objects can cause inconsistent performance.
+
+Recommendations:
+
+* **Do** consider caching large objects that are frequently used so that they don't need to be reallocated each time they're needed.
+* **Do** pool buffers by using an [`ArrayPool<T>`](/dotnet/api/system.buffers.arraypool-1) to store large arrays.
+* **Do not** allocate many, short-lived large objects on [hot code paths](#hot).
+
+Memory issues like the preceding can be diagnosed by reviewing garbage collection (GC) stats in [PerfView](https://github.com/Microsoft/perfview) and examining:
+
+* Garbage collection pause time.
+* What percentage of the processor time is spent in garbage collection.
+* How many garbage collections are generation 0, 1, and 2.
+
+For more information, see [Garbage Collection and Performance](/dotnet/standard/garbage-collection/performance).
+
+## Optimize Data Access
+
+Interactions with a data store or other remote services are often the slowest parts of an ASP.NET Core app. It's important to make sure that data is read and written efficiently. In addition to ensuring data access is asynchronous, some best practices include:
+
+* **Do not** retrieve more data than is necessary. Write queries to return just the data that is necessary for the current HTTP request.
```

Comment:
Nit-pick - I just noticed that the other do/do-not sections begin with the line "Recommendations:" and this one does not.